projects/packages/forms/src/contact-form/css/grunion.css

```diff
@@ -685,36 +700,43 @@ on production builds, the attributes are being reordered, causing side-effects
 	background: currentColor;
 	border-radius: 50%;
 	content: '';
-	height: calc(var(--jetpack--contact-form--font-size) / 2);
+
+	/* Inherit the font size from the parent element. This allows the input to scale with the font size
+	   set from global styles or from block styles */
+	font-size: inherit;
+	height: calc(1em / 2);
+	width: calc(1em / 2);
 	margin-left: 50%;
 	margin-top: 50%;
 	position: absolute;
 	transform: translate(-50%, -50%);
-	width: calc(var(--jetpack--contact-form--font-size) / 2);
 }
 
 .contact-form .grunion-field-wrap.grunion-field-checkbox-multiple-wrap.is-style-button-wrap .contact-form-field,
-.contact-form .grunion-field-wrap.is-style-button-wrap .grunion-radio-label {
+.contact-form .grunion-field-wrap.grunion-field-radio-wrap.is-style-button-wrap .contact-form-field {
 	display: inline-flex;
 	align-items: center;
 	padding: var(--jetpack--contact-form--button-outline--padding);
+}
+
+.contact-form :where(.grunion-field-wrap.grunion-field-checkbox-multiple-wrap.is-style-button-wrap .contact-form-field),
+.contact-form :where(.grunion-field-wrap.grunion-field-radio-wrap.is-style-button-wrap .contact-form-field) {
 	background: var(--jetpack--contact-form--button-outline--background-color);
 	border: var(--jetpack--contact-form--button-outline--border);
 	border-radius: var(--jetpack--contact-form--button-outline--border-radius);
-	color: var(--jetpack--contact-form--button-outline--text-color);
```

Comment:
When no color is set for the Option block in global styles, I think this removal results in the text color being black on the site frontend when using the `button` style:

<img width="1026" alt="image" src="https://github.com/user-attachments/assets/342dbc5d-e453-4acb-806a-f3e6aa57ac69" />

(It's working fine when I manually set a text color for the Option block in global styles)

Reply:
That was probably accidental, not sure why I would've deleted that line. 🤔

Reply:
It looks like another similar issue is that the block styles are applying inconsistently to the input. Block styles are only applying to the option label, whereas global styles are applying to the label and input.

Here I've set the global style color for the option to orange, and the block style color for the orange to green:
<img width="116" alt="Screenshot 2025-05-01 at 2 12 37 pm" src="https://github.com/user-attachments/assets/92bfb07e-f2f8-4a1a-8241-0f4b09f6c1dc" />

I need to double check trunk to make sure I've got this right (I think with the button style, both the label and input should receive a color).

Reply:
I think I've fixed all the issues in this thread now 🤞